_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+
+Open `projects/mfe-profile/src/app/profile/profile.component.ts` and edit the component to add the two public properties and include the `OktaAuthStateService` in the constructor:
+
+```ts
+import { Component, OnInit } from '@angular/core';
+import { OktaAuthStateService } from '@okta/okta-angular';
+import { filter, map } from 'rxjs';
+
+@Component({
+  selector: 'app-profile',
+  templateUrl: './profile.component.html',
+  styles: []
+})
+export class ProfileComponent {
+  public profile$ = this.oktaStateService.authState$.pipe(
+    filter(state => !!state && !!state.isAuthenticated),
+    map(state => state.idToken?.claims)
+  );
+
+  public date$ = this.oktaStateService.authState$.pipe(
+    filter(state => !!state && !!state.isAuthenticated),
+    map(state => (state.idToken?.claims.auth_time as number) * 1000),
+    map(epochTime => new Date(epochTime)),
+  );
+
+  constructor(private oktaStateService: OktaAuthStateService) { }
+}
+```
+
+Next, open the corresponding template file and replace the existing code with the following:
+
+{% raw %}
+```html
+<h3 class="text-xl mb-6">Your Profile</h3>
+
+<div *ngIf="profile$ | async as profile">
+  <p>Name: <span class="font-semibold">{{profile.name}}</span></p>
+  <p class="my-3">Email: <span class="font-semibold">{{profile.email}}</span></p>
+  <p>Last signed in at <span class="font-semibold">{{date$ | async | date:'full'}}</span></p>
+</div>
+```
+{% endraw %}
+
+Try running the `mfe-profile` app by itself by running `ng serve mfe-profile --open` in the terminal. Notice when we navigate to the `/profile` route, we see a console error. We added Okta into the `shell` application, but now we need to turn the `mfe-profile` application into a micro-frontend and share the authenticated state. Stop serving the application so we're ready for the next step.
+
+## Turn the Angular application into a micro-frontend remote
+
+We want to use the schematic from `@angular-architects/module-federation` to turn the `mfe-profile` application into a micro-frontend and add the necessary configuration. We'll use port 4202 for this application. Add the schematic by running the following command in the terminal:
+
+```shell
+ng add @angular-architects/module-federation --project mfe-profile --port 4202
+```
+
+This schematic does the following:
+1. Updates the project's `angular.json` config file to add the port for the application and updates the builder to use a custom Webpack builder
+2. Creates the `webpack.config.js` files and scaffolds out default configuration for Module Federation
+
+First, let's add the new micro-frontend to the `shell` application by updating the configuration in `projects/mfe-profile/webpack.config.js`. In the middle of the file, there's a property for `plugins` with commented-out code. We need to finish configuring that. Since this application is a remote, we'll update the snippet of code under the comment:
```

Comment:
```suggestion
First, let's add the new micro frontend to the `shell` application by updating the configuration in `projects/mfe-profile/webpack.config.js`. In the middle of the file, there's a property for `plugins` with commented-out code. We need to finish configuring that. Since this application is a remote, we'll update the snippet of code under the comment:
```
hyphen removed